papers/scipy/scipy.rst

```diff
@@ -0,0 +1,1706 @@
+texts on digital signal processing explain the algorithm (e.g. Section
+7.7 of Oppenheim and Schafer [OS]_). The method is implemented in ``scipy.signal``
+by the function ``remez``.
+
+As an example, we'll design a bandpass filter for a signal
+with a sampling rate of 2000 Hz using ``remez``.
+For this filter, we want the stop bands to be [0, 250] and [700, 1000],
+and the pass band to be [350, 550].  We'll leave the behavior outside
+these bands unspecified, and see what ``remez`` gives us.
+We'll use 31 taps.
+
+.. code-block:: python
+
+    fs = 2000
+    bands = [0, 250, 350, 550, 700, 0.5*fs]
+    desired = [0, 1, 0]
+
+    numtaps = 31
+
+    taps = remez(numtaps, bands, desired, Hz=fs)
+
+The frequency response of this filter is the curve labeled ``(a)``
+in Fig. :ref:`fig-remez-example-31taps`.
+
+
+To reduce the ripple in the pass band while using the same filter length,
+we'll adjust the weights, as follows:
+
+.. code-block:: python
+
+    weights = [1, 25, 1]
+    taps2 = remez(numtaps, bands, desired, weights, Hz=fs)
+
+The frequency response of this filter is the curve labeled ``(b)``
+in Fig. :ref:`fig-remez-example-31taps`.
+
+.. figure:: figs/remez_example_31taps.pdf
+
+   Frequency response of bandpass filters designed using
+   ``scipy.signal.remez``.  The stop bands are [0, 250] and [700, 1000],
+   and the pass band is [350, 550].  The shaded regions are the "don't care"
+   intervals where the desired behavior of the filter is unspecified.
+   The curve labeled `(a)` uses the default weights--each band
+   is given the same weight.  For the curve labeled `(b)`,
+   `weight = [1, 25, 1]` was used.
+
+   :label:`fig-remez-example-31taps`
+
+It is recommended to always check the frequency response of a filter
+designed with ``remez``.  Figure :ref:`fig-remez-example-47taps` shows
+the frequency response of the filters when the number of taps is
+increased from 31 to 47.  The ripple in the pass and stop bands is
+decreased, as expected, but the behavior of the filter in the
+interval [550, 700] might be unacceptable.  This type of behavior
+is not unusual for filters designed with ``remez`` when there
+are intervals with unspecified desired behavior.
+
+.. figure:: figs/remez_example_47taps.pdf
+
+   This plot shows the results of the same
+   calculation that produced Figure :ref:`fig-remez-example-31taps`,
+   but the number of taps has been increased from 31 to 47.
+   Note the possibly undesirable behavior of the filter in the
+   transition interval [550, 700].
+
+   :label:`fig-remez-example-47taps`
+
+In some cases, the exchange algorithm implemented in ``remez`` can fail
+to converge.  This can happen, for example, when the transition between
+bands or the bands themselves are too small.  Failure is more likely when
+the number of taps is large (i.e. greater than 1000).  When the algorithm
+fails to converge, ``remez`` raises an exception, so the failure can be
```

Comment:
This sentence could be left out.

Reply:
Looking at the rest of the section, the discussion on failures can be shortened significantly I think. Perhaps just say that for >1000 taps there may be issues and recommend to check the frequence response.

Reply:
OK, shortened.